dipy/segment/energy_mrf.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import division, print_function, absolute_import
+import numpy as np
+
+
+def total_energy(masked_image, masked_segmentation,
+                 mu, var, index, label, beta):
+
+    energytotal = log_likelihood(masked_image, mu, var, index, label)
+    energytotal += gibbs_energy(masked_segmentation, index, label, beta)
+
+    return energytotal
+
+
+def log_likelihood(img, mu, var, index, label):
+
+    loglike = ((img[index] - mu[label]) ** 2) / (2 * var[label])
+
+    loglike += np.log(np.sqrt(var[label]))
+
+    return loglike
+
+
+def gibbs_energy(seg, index, label, beta):
+
+    energy = 0
+
+    if label == seg[index[0] - 1, index[1], index[2]]:
```

Comment:
I think you need to check if you are at the boundary of the image and do not accumulate the energy in that case, otherwise this will wrap to the other side in Python (not the desired behavior) and may cause a memory access violation in Cython (I think we will definitely need to cythonize)